v4/opt/norm/push_down.opt

```diff
@@ -13,9 +13,24 @@
     (Filters (RemoveListItem $list $condition))
 )
 
+[PushDownSelectJoinSelectLeft, Normalize]
+(Select
+    $input:(Join (Select $inputInner:* (Filters [ ... $filterInner:* ... ])) $right:* $on:*)
```

Comment:
The `[ ... $filterInnder:* ... ]` pattern will match the first item in the list (because you used * to match). But what if there is more one item in the list? Those other items would get discarded by the replace pattern.

I think I see why you added this pattern - my default pushdown pattern would wrap a `Select` with a `Select`, and that pattern avoids doing that. However, I think it'd be better to create a pattern similar to this:
```
[MergeSelectSelect, Normalize]
(Select
  (Select $input $innerFilter:*)
  $outerFilter:*
)
=>
(Select
  $input
  (ConcatFilterConditions $outerFilter $innerFilter)
)
```

Because of the way the factory works, this is going to be fairly efficient, as we don't actually construct the outer `Select`. Also, this pattern will match in many other useful cases besides Pushdown.